src/Testcontainers/Clients/TestcontainersClient.cs

```diff
@@ -339,7 +339,7 @@ public async Task<string> BuildAsync(IImageFromDockerfileConfiguration configura
         var cachedImages = await Image.GetAllAsync(filters, ct)
           .ConfigureAwait(false);
 
-        var repositoryTags = new HashSet<string>(cachedImages.SelectMany(image => image.RepoTags));
+        var repositoryTags = new HashSet<string>(cachedImages.SelectMany(image => image.RepoTags ?? Array.Empty<string>()));
```

Comment:
```suggestion
        var repositoryTags = new HashSet<string>(cachedImages.Where(image => image.RepoTags != null).SelectMany(image => image.RepoTags));
```

Reply:
Here's my thoughts:
- LINQ version optimizes the error path at cost of the happy path
- Nullable can't infer the null check from Where to the SelectMany, so whenever you turn on NRTs (which might have caught this at compile time!), you will get an NRT error in the SelectMany

However I'm open to anything, it's up to you to chose what you prefer.

Reply:
@HofmeisterAn Just give me a short notice

Reply:
No, there is no need to change it. Your explanation is reasonable, and there isn't much difference (it's just my personal preference; I don't particularly like the null-coalescing operator and generally prefer LINQ).